### ISSUE TITLE
[Bug]: error `MODULE_NOT_FOUND` on node 20+ but works on node 18

### ISSUE THREAD

**Issue by shulaoda** [napi-rs/napi-rs#2215]
In my project, I am experiencing inconsistent behavior depending on the node version. Specifically, it works correctly when running on `node 18`. However, when I try to run the same code on `node 20+` , I encounter a `MODULE_NOT_FOUND` error.

```rust
#![deny(clippy::all)]

use napi::Either;

#[macro_use]
extern crate napi_derive;

#[napi]
pub fn print(value: Either<bool, String>) {
  println!("{:?}", value);
}

```
<img width="495" alt="image" src="https://github.com/user-attachments/assets/347f3ee5-1348-489f-8a11-cafaa98be4eb">

### Reproduce steps
1.  git clone this [repo](https://github.com/shulaoda/repro-cases/tree/either-issue) on macos machine
2. `pnpm i`
3. `pnpm run build`
4. `pnpm run test`

**Comment by shulaoda** [comment]
Related to https://github.com/rolldown/rolldown/pull/1898, https://github.com/rolldown/rolldown/pull/1892

**Comment by shulaoda** [comment]
System:
  OS: macOS 14.4.1
  CPU: (11) x64 Apple M3 Pro
  Memory: 18.15 MB / 18.00 GB
  Shell: 5.9 - /bin/zsh

**Comment by shulaoda** [comment]
This issue seems to have been fixed in the 3.0 alpha version.

**Comment by shulaoda** [comment]
This doesn't seem to be related to `napi-rs`, it's a configuration issue with my computer environment.

**Comment by shulaoda** [comment]
It seems that this issue arose because I migrated from an Intel `x86_64` Mac to a new `ARM64` Mac, and the `Rust toolchain` was also migrated in the process. I did not encounter any errors or unusual behavior during this transition and have been using it without issues until now.
I discovered that the `napi-rs` generates a `.node` file using a function called `getHostTargetTriple`, which executes the command `rustc -vV` to determine the architecture type. This has caused an inconsistency between the architecture information retrieved by `Rust` and the platform information retrieved by `Node.js`.
